uportal-war/src/main/java/org/apereo/portal/security/provider/cas/CasAssertionSecurityContext.java

```diff
@@ -167,12 +205,28 @@ protected void copyAssertionAttributesToUserAttributes(Assertion assertion) {
             log.debug("Adding attribute '{}' from Assertion with value '{}'; runtime type of value is {}",
                     attrEntry.getKey(), attrEntry.getValue(), attrEntry.getValue().getClass().getName());
 
+            // Check for credential
+            if (decryptCredentialToPassword && key != null && cipher != null && attrEntry.getKey().equals(CREDENTIAL_KEY)) {
+                try {
+                    final String encPwd = (String) (attrEntry.getValue() instanceof List
+                            ? ((List) attrEntry.getValue()).get(0) : attrEntry.getValue());
+                    byte[] cred64 = DatatypeConverter.parseBase64Binary(encPwd);
+                    cipher.init(Cipher.DECRYPT_MODE, key);
+                    final byte[] cipherData = cipher.doFinal(cred64);
+                    final Object pwd = new String(cipherData);
+                    attributes.put(PASSWORD_KEY, Collections.singletonList(pwd));
+                } catch (Exception e) {
+                    log.warn("Cannot decipher credential", e);
+                    e.printStackTrace();
```

Comment:
I'm thinking we shouldn't include this line.  The previous line will write the stack trace to `portal.log` (which is great);  this one will write the same stack trace to `catalina.out`, but without the message.  I can't think of anywhere else we do that.